jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -366,9 +412,77 @@ public String getRFC6265SetCookie()
             buf.append("; Secure");
         if (_httpOnly)
             buf.append("; HttpOnly");
+        if (_sameSite != null)
+        {
+            buf.append("; SameSite=");
+            buf.append(_sameSite.toString());
+        }
+
         return buf.toString();
     }
 
+    private static boolean isHttpOnlyInComment(String comment)
+    {
+        return comment != null && comment.contains(HTTP_ONLY_COMMENT);
+    }
+
+    private static SameSite getSameSiteFromComment(String comment)
+    {
+        if (comment != null)
+        {
+            if (comment.contains(SAME_SITE_NONE_COMMENT))
+            {
+                return SameSite.NONE;
+            }
+            if (comment.contains(SAME_SITE_LAX_COMMENT))
+            {
+                return SameSite.LAX;
+            }
+            if (comment.contains(SAME_SITE_STRICT_COMMENT))
+            {
+                return SameSite.STRICT;
+            }
+        }
+
+        return null;
+    }
+
+    private static String getCommentWithoutFlags(String comment)
+    {
+        if (comment == null)
+        {
+            return null;
+        }
+
+        String strippedComment = comment.trim();
+
+        if (isHttpOnlyInComment(strippedComment))
```

Comment:
I think that rather than look and then stripping, you may as well just call the 4 strip methods one after the other.  If they don't find then then don't strip, so we save a string search for each.